slicingTool/runProjectSlicingTool.sh

```diff
@@ -26,14 +34,23 @@ if [[ -n $DB && -n $USER && -n $PASS ]]
 then
     if [[ $USER =~ "curator" ]]
     then
-	echo Dropping $DB ...
-	mysql -u$USER -p$PASS -e "drop database if exists $DB"
-#	mysql -u$USER -p$PASS -e "drop database if exists ${DB}_myisam"
-#	mysql -u$USER -p$PASS -e "create database ${DB}_myisam"
-#	mysql -u$USER -p$PASS -e "drop database if exists gk_central"
-#	mysql -u$USER -p$PASS -e "create database gk_central"
+        # Attempts to find database $DB and if successful (mysqlshow returns a zero exit status) it is backed up
+        # before being dropped
+        mysqlshow $DB > /dev/null 2>&1
```

Comment:
Does `mysqlshow` require user/host/password information?

Reply:
Yeah, it may.  I think I have a .my.cnf file set up so it is able to work without command line credentials.  I'll add user and password (I don't think it needs host) until we implement a way for release to use files for credentials.

Reply:
Also, why discard the output? If there's a problem, maybe the output would be useful in debugging it?

Reply:
Well, the idea was that `mysqlshow` was just being used as a test to see if the database exists or not.  If it does, the command is successful and the exit status is 0.  If it doesn't the command fails and the exit status is non-zero.  I was more interested in the exit status as a flag for seeing if it exists and needs backing up.  If it succeeds, I don't really care about the output showing the database name.  

If it fails, it's trickier.  If it fails because the db doesn't exist, then that's not really an error for how I'm using it, but if it fails for another reason then it would be useful to know why.

Is there a simpler way to check if a db exists and should be backed up?

Reply:
A different way would be to connect to the database and execute a query, something that should always succeed, like querying for the Reactome version. Not sure if that's simpler. Again, that could fail for any/all of these reasons:
 - the database does not exist
 - the table does not exist
 - the query is malformed
 - the connection information is incorrect
 - ...probably others.

I didn't know about `mysqlshow` before, but it seems like it's slightly simpler than executing a query.

The way you are using it here, it will print a list of tables in the `$DB`-database and then exit. If there is an error, it will print an error. I would _strongly_ recommend saving the stderr and stdout. You can dump them both to a file called `.mysqlshow.results` but I think having that file would be useful, in case there is an error. Otherwise, all the user gets is exit code != 0. Which could mean...?

Reply:
Okay, I'll save the output and error to a file.

Maybe `mysql -u $USER -p$PASS -e 'use $DB'` would be better?  It still will print a bit to STDOUT but not all the table names.

Reply:
Further testing shows `mysqlshow` might not be safe. Try using a database name that has the same _prefix_ as a _real_ database. For example:

```
$ mysqlshow -uUSER -p MyDB_SOMEBLAHBLAH
Enter password: 
Wildcard: MyDB_SOMEBLAHBLAH
+-----------+
| Databases |
+-----------+
+-----------+
$ echo $?
0
```

Assuming that `MyDB` is a _real_ database and `MyDB_SOMEBLAHBLAH` doesn't exist, it looks like `mysqlshow` does a wildcard search. It shows nothing, but exit code is still 0.

Reply:
If you run `mysqlshow` with a database name that is not a prefix OR a suffix to a real database, _then_ you get an error:
```
$ mysqlshow -uUSER -p X
Enter password: 
mysqlshow: Unknown database 'X'
$ echo $?
1
```

In conclusion, I am not sure that `mysqlshow` is the best way to do this.

Reply:
...yes, perhaps `... -e 'use $DB'` is better.

Reply:
Yup, makes sense.  Using `-e 'use $DB' now`